### PR TITLE
New Backbones: EffNetV2 family and Bottleneck Transformers

### PR DESCRIPTION
**Pull request by DuncanZauss**
Added backbones:
- EffNetV2 family backbones (https://arxiv.org/pdf/2104.00298.pdf)
- EffNetV2 family with a stride of 16
- BotNet (https://arxiv.org/pdf/2101.11605.pdf)

### CODE REVIEW

**Review by svenkreiss — COMMENTED** [review]
Thanks @DuncanZauss for bringing effnet and bottleneck transformers to openpifpaf.
I have a few comments below. I am mostly worried about licensing more than anything else.

File: openpifpaf/network/basenetworks.py [hunk 1]
```diff
@@ -496,3 +499,207 @@ def cli(cls, parser: argparse.ArgumentParser):
     @classmethod
     def configure(cls, args: argparse.Namespace):
         cls.pretrained = args.squeezenet_pretrained
+
+
+class EffNetV2S(BaseNetwork):
```
[the remark this answers is not in the document]
Comment (DuncanZauss) [reply openpifpaf/network/basenetworks.py]: I have now changed it so that the configuration is provided in factory.py, just like it is done for the ShuffleNets.

File: openpifpaf/network/factory.py [hunk 2]
```diff
@@ -34,7 +34,9 @@
     basenetworks.ShuffleNetV2,
     basenetworks.ShuffleNetV2K,
     basenetworks.SqueezeNet,
-])
+    basenetworks.BotNet,
+])  # TODO add EffNetV2, once they are cleaned up and have CLI options
```
[the remark this answers is not in the document]
Comment (DuncanZauss) [reply openpifpaf/network/factory.py]: Done

File: setup.py [hunk 3]
```diff
@@ -60,6 +60,7 @@
         'torchvision>=0.8.2',
         'pillow<8.3',  # upper limit for torchvision 0.10.0 compatibility
         'dataclasses; python_version<"3.7"',
+        'einops>=0.3',  # required for BotNet
```
[the remark this answers is not in the document]
Comment (DuncanZauss) [reply setup.py:63]: I have added einops to the new `backbones` requirement.

File: openpifpaf/network/bottleneck_transformer.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
+import torch
+from torch import nn, einsum
+
+from einops import rearrange
+
+# translated from tensorflow code
+# https://gist.github.com/aravindsrinivas/56359b79f0ce4449bcb04ab4b56a57a2
```
[the remark this answers is not in the document]
Comment (DuncanZauss) [reply openpifpaf/network/bottleneck_transformer.py]: Both files have the MIT license, I have added the licensing to the docs folder and refer to those files, just like David did with his new backbones.
In case you also want to have a look:
EffNetV2 license: https://github.com/d-li14/efficientnetv2.pytorch/blob/main/LICENSE
BotNet license: https://github.com/lucidrains/bottleneck-transformer-pytorch/blob/main/LICENSE

File: LICENSE.save [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Copyright 2019-2021 by Sven Kreiss and contributors. All rights reserved.
```
Comment (svenkreiss) [inline LICENSE.save:1]: accidental commit?

File: openpifpaf/network/basenetworks.py.orig [hunk 6]
```diff
@@ -0,0 +1,630 @@
+import argparse
```
Comment (svenkreiss) [inline openpifpaf/network/basenetworks.py.orig:1]: accidental commit?

File: openpifpaf/network/factory.py.orig [hunk 7]
```diff
@@ -0,0 +1,294 @@
+import argparse
```
Comment (svenkreiss) [inline openpifpaf/network/factory.py.orig:1]: accidental commit?

File: setup.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (svenkreiss) [reply setup.py:63]: please remove it here

**Review by svenkreiss — COMMENTED** [review]
Thanks @DuncanZauss !
Can you take a look at the diff? There seem to be some unintended files in this PR.

File: LICENSE.save (hunk 5 above)
Comment (DuncanZauss) [reply LICENSE.save]: Thanks for pointing that out! I accidentally committed those files, that came from the merge tool.
I deleted the files.

**Comment by svenkreiss** [comment]
Thanks @DuncanZauss !
